src/Raven.Server/Smuggler/Documents/StreamDestination.cs

```diff
@@ -108,246 +114,396 @@ public IIndexActions Indexes()
                 }
 
-                _writer.WritePropertyName(nameof(databaseRecord.Client));
-                WriteClientConfiguration(databaseRecord.Client);
-            }
-
-            private void WriteClientConfiguration(ClientConfiguration clientConfiguration)
-            {
-                if (clientConfiguration == null)
+                if (databaseRecordItemType.Contain(DatabaseRecordItemType.Revisions))
                 {
-                    _writer.WriteNull();
-                    return;
+                    _writer.WriteComma();
+                    _writer.WritePropertyName(nameof(databaseRecord.Revisions));
+                    WriteRevisions(databaseRecord.Revisions);
                 }
 
-                _writer.WriteStartObject();
-
-                _writer.WritePropertyName(nameof(clientConfiguration.Etag));
-                _writer.WriteInteger(clientConfiguration.Etag);
-                _writer.WriteComma();
-
-                _writer.WritePropertyName(nameof(clientConfiguration.Disabled));
-                _writer.WriteBool(clientConfiguration.Disabled);
-
-                if (clientConfiguration.MaxNumberOfRequestsPerSession.HasValue)
+                if (databaseRecordItemType.Contain(DatabaseRecordItemType.Expiration))
                 {
                     _writer.WriteComma();
-                    _writer.WritePropertyName(nameof(clientConfiguration.MaxNumberOfRequestsPerSession));
-                    _writer.WriteInteger(clientConfiguration.MaxNumberOfRequestsPerSession.Value);
+                    _writer.WritePropertyName(nameof(databaseRecord.Expiration));
+                    WriteExpiration(databaseRecord.Expiration);
                 }
 
-                if (clientConfiguration.PrettifyGeneratedLinqExpressions.HasValue)
+                if (databaseRecordItemType.Contain(DatabaseRecordItemType.Client))
                 {
                     _writer.WriteComma();
-                    _writer.WritePropertyName(nameof(clientConfiguration.PrettifyGeneratedLinqExpressions));
-                    _writer.WriteBool(clientConfiguration.PrettifyGeneratedLinqExpressions.Value);
+                    _writer.WritePropertyName(nameof(databaseRecord.Client));
+                    WriteClientConfiguration(databaseRecord.Client);
                 }
 
-                if (clientConfiguration.ReadBalanceBehavior.HasValue)
+                if (databaseRecordItemType.Contain(DatabaseRecordItemType.Sorters))
                 {
                     _writer.WriteComma();
-                    _writer.WritePropertyName(nameof(clientConfiguration.ReadBalanceBehavior));
-                    _writer.WriteString(clientConfiguration.ReadBalanceBehavior.Value.ToString());
+                    _writer.WritePropertyName(nameof(databaseRecord.Sorters));
+                    WriteSorters(databaseRecord.Sorters);
                 }
 
-                _writer.WriteEndObject();
+                switch (authorizationStatus)
```

Comment:
Why do we have a check here?

Reply:
We are writing here certificates, password etc.